### PR TITLE
A round of cleanup in the execution engine

### PR DESCRIPTION
**Pull request by lptr**
Some minor improvements around the execution engine.

### CODE REVIEW

File: subprojects/execution/src/main/java/org/gradle/internal/execution/UnitOfWork.java [hunk 1]
```diff
@@ -55,20 +53,41 @@
 
     void visitInputProperties(InputPropertyVisitor visitor);
 
+    interface Property {
+        String getName();
+    }
+
+    interface InputProperty extends Property {
```
Comment (wolfs) [inline subprojects/execution/src/main/java/org/gradle/internal/execution/UnitOfWork.java]: We have been moving towards using primitive types for the `visit` method arguments. Do we want to move towards using own interfaces for the visit methods now? What is the motivation behind this - so what is the extra functionality we want to support for different kinds of properties without using suppliers?

File: subprojects/execution/src/test/groovy/org/gradle/internal/execution/steps/StepSpec.groovy [hunk 2]
```diff
@@ -19,10 +19,11 @@ package org.gradle.internal.execution.steps
```
Comment (wolfs) [inline subprojects/execution/src/test/groovy/org/gradle/internal/execution/steps/StepSpec.groovy]: Nit: Double new line.

File: subprojects/execution/src/main/java/org/gradle/internal/execution/UnitOfWork.java (hunk 1 above)
Comment (lptr) [reply subprojects/execution/src/main/java/org/gradle/internal/execution/UnitOfWork.java]: We already do fingerprinting as a `Supplier`. The other thing we'll need eventually is validation. We could expose that as well as a `Runnable`, too, but it's nicer this way and easier to document IMO.

Comment (lptr) [reply subprojects/execution/src/main/java/org/gradle/internal/execution/UnitOfWork.java]: Taking a step back...

This API is taking shape. I think it's easier to make changes to it this way. We can revisit the decision here (and the whole API) once the functionality is complete.

Comment (wolfs) [reply subprojects/execution/src/main/java/org/gradle/internal/execution/UnitOfWork.java]: I also agree that the API is easier to change with own argument types for the visitors. Let's try this out and see where it takes us.

Comment (lptr) [reply subprojects/execution/src/main/java/org/gradle/internal/execution/UnitOfWork.java]: After thinking a bit more about this I ended up discarding the introduction of `UnitOfWork.*Property` types for now. Adding validation didn't require them in #9991, so let's revisit this again if and when it becomes a priority.

**Comment by lptr** [comment]
Closing in favor of #9991 that includes all changes except for the `UnitOfWork.*Property` types.

**Pull request closed without merging**